src/main/java/com/graphhopper/converter/resources/ConverterResourceGisgraphy.java

```diff
@@ -0,0 +1,199 @@
+			LOGGER.error("There was an issue with the target "
+					+ target.getUri() + " the provider returned: "
+					+ status.code + " - " + status.message);
+			throw new BadRequestException(
+					"error deserializing geocoding feed");
+		}
+	}
+
+	protected void processFeedException(WebTarget target, Status status) {
+		
+	}
+
+	private void checkParameters(String query, boolean reverse, boolean autocomplete, String point) {
+		super.checkInvalidParameter(reverse,query,point);
+		if (reverse && autocomplete){
+			throw new BadRequestException("autocomplete is not available in reverse geocoding request, set reverse or autocomplete to false but not both");
+		}
+
+	}
+
+	private WebTarget buildGeocodingTarget(String query, String lat,
+			String lng, String radius, String country, int limit) {
+		WebTarget target = jerseyClient.target(geocodingUrl)
+				.queryParam(ADDRESS_PARAMETER, query)
+				.queryParam(FORMAT_PARAMETER, DEFAULT_FORMAT);
+		if (!apiKey.isEmpty()) {
+			target = target.queryParam(APIKEY_PARAMETER, apiKey);
+		}
+		if (lat!=null && !lat.isEmpty()) {
+			target = target.queryParam(LAT_PARAMETER, lat);
+		}
+		if (lng != null && !lng.isEmpty()) {
+			target = target.queryParam(LONG_PARAMETER, lng);
+		}
+		if (!radius.isEmpty()) {
+			target = target.queryParam(RADIUS_PARAMETER, radius);
+		}
+		if (!country.isEmpty()) {
+			target = target.queryParam(COUNTRY_PARAMETER, country);
+		}
+		if (limit >0) {
+			target = target.queryParam(GEOCODING_LIMIT_PARAMETER, limit);
+		}
+		return target;
+	}
+
+	private WebTarget buildReverseGeocodingTarget(String query, String lat,
+			String lng, String radius, String country, int limit) {
+		WebTarget target = jerseyClient.target(reverseGeocodingUrl).queryParam(
+				FORMAT_PARAMETER, DEFAULT_FORMAT);
+		if (!apiKey.isEmpty()) {
+			target = target.queryParam(APIKEY_PARAMETER, apiKey);
+		}
+		if (lat!=null && !lat.isEmpty()) {
+			target = target.queryParam(LAT_PARAMETER, lat);
+		}
+		if (lng!=null && !lng.isEmpty()) {
+			target = target.queryParam(LONG_PARAMETER, lng);
+		}
+		return target;
+	}
+
+	private WebTarget buildAutocompleteTarget(String query, String lat,
+			String lng, String radius, String country, int limit) {
+		WebTarget target = jerseyClient.target(searchURL)
+				.queryParam(SEARCH_QUERY_PARAMETER, query)
+				.queryParam(FORMAT_PARAMETER, DEFAULT_FORMAT)
+				.queryParam("suggest", "true");
+		if (!apiKey.isEmpty()) {
+			target = target.queryParam(APIKEY_PARAMETER, apiKey);
+		}
+		if (lat!=null && !lat.isEmpty()) {
+			target = target.queryParam(LAT_PARAMETER, lat);
+		}
+		if (lng!=null && !lng.isEmpty()) {
+			target = target.queryParam(LONG_PARAMETER, lng);
+		}
+		if (!radius.isEmpty()) {
+			target = target.queryParam(RADIUS_PARAMETER, radius);
+		}
+		if (!country.isEmpty()) {
+			target = target.queryParam(COUNTRY_PARAMETER, country);
+		}
+		if (limit >0) {
+			target = target.queryParam(SEARCH_LIMIT_PARAMETER, limit);
+		}
+		return target;
+	}
```

Comment:
Would it be possible to reduce the code duplicateion for the 3 buildTarget methods?

All 3 use these lines:
```
if (!apiKey.isEmpty()) {
			target = target.queryParam(APIKEY_PARAMETER, apiKey);
		}
		if (lat!=null && !lat.isEmpty()) {
			target = target.queryParam(LAT_PARAMETER, lat);
		}
		if (lng!=null && !lng.isEmpty()) {
			target = target.queryParam(LONG_PARAMETER, lng);
		}
``` 

2 use these lines:
```
		if (!radius.isEmpty()) {
			target = target.queryParam(RADIUS_PARAMETER, radius);
		}
		if (!country.isEmpty()) {
			target = target.queryParam(COUNTRY_PARAMETER, country);
		}
```

Reply:
Done.